### PR TITLE
[DRAFT] feat: Use Verificient Onboarding API for instructor dashboard if waffle flag is enabled

### PR DESCRIPTION
**Pull request by alangsto**
* Basic testing for use of onboarding API
* Added function to gather all paginated data from proctoring provider onboarding endpoint
* Updated frontend to only use specific filters if onboarding API is being used

### CODE REVIEW

File: edx_proctoring/backends/rest.py [hunk 1]
```diff
@@ -349,6 +353,9 @@ def get_onboarding_profile_info(self, course_id, **kwargs):
         data = response.json()
         return data
 
+    def translate_filter_status(self, filter_status):
```
Comment (alangsto) [inline edx_proctoring/backends/rest.py:356]: I moved the translation of statuses here, should we do that for other backend specific translations?

**Review by MichaelRoytman — COMMENTED** [review]
I left a few comments. It looks good to me! Thank you for all your hard work on this.

File: edx_proctoring/statuses.py [hunk 2]
```diff
@@ -311,12 +311,30 @@ class VerificientOnboardingProfileStatus:
         pending: ProctoredExamStudentAttemptStatus.submitted
     }
 
+    filter_status_mapping = {
+        InstructorDashboardOnboardingAttemptStatus.not_started: no_profile,
+        ProctoredExamStudentAttemptStatus.submitted: pending,
```
[the remark this answers is not in the document]
Comment (alangsto) [reply edx_proctoring/statuses.py:316]: I agree, that would make this more readable. Will update!

File: edx_proctoring/static/proctoring/js/views/proctored_exam_onboarding_view.js [hunk 3]
```diff
@@ -16,6 +16,13 @@ edx = edx || {};
         'rejected',
         'error'
     ];
+    onboardingProfileAPIStatuses = [
```
[the remark this answers is not in the document]
Comment (alangsto) [reply edx_proctoring/static/proctoring/js/views/proctored_exam_onboarding_view.js:19]: Yes, thanks for catching that

File: edx_proctoring/static/proctoring/js/views/proctored_exam_onboarding_view.js [hunk 4]
```diff
@@ -59,6 +66,8 @@ edx = edx || {};
             this.collection = new edx.instructor_dashboard.proctoring.ProctoredExamOnboardingCollection();
             this.templateUrl = '/static/proctoring/templates/student-onboarding-status.underscore';
             this.courseId = this.$el.data('course-id');
+            this.enableProctoringApi = this.$el.data('use-onboarding-api');
```
[the remark this answers is not in the document]
Comment (alangsto) [reply edx_proctoring/static/proctoring/js/views/proctored_exam_onboarding_view.js:69]: This is part of an edx-platform PR that I need to make. In the past this is how we've passed waffle switches/flags from the backend to the frontend

File: edx_proctoring/static/proctoring/js/views/proctored_exam_onboarding_view.js [hunk 5]
```diff
@@ -59,6 +66,8 @@ edx = edx || {};
...
+            this.enableProctoringApi = this.enableProctoringApi && this.enableProctoringApi.toLowerCase() === 'true';
```
[the remark this answers is not in the document]
Comment (alangsto) [reply edx_proctoring/static/proctoring/js/views/proctored_exam_onboarding_view.js:70]: I'll test it out!

File: edx_proctoring/static/proctoring/templates/student-onboarding-status.underscore [hunk 6]
```diff
@@ -91,14 +91,15 @@
                 </ul>
                 <div class="clearfix"></div>
             </div>
+            <% var isOnboardingItems = onboardingItems.length !== 0 %>
```
[the remark this answers is not in the document]
Comment (alangsto) [reply edx_proctoring/static/proctoring/templates/student-onboarding-status.underscore]: This was one of the things I forgot to remove earlier - it's removed now 🙂

**Pull request merged**